ext/standard/array.c

```diff
@@ -982,7 +982,7 @@ static int php_array_user_compare(const void *a, const void *b) /* {{{ */
 		zval_ptr_dtor(&retval);
 		zval_ptr_dtor(&args[1]);
 		zval_ptr_dtor(&args[0]);
-		return ret < 0 ? -1 : ret > 0 ? 1 : 0;
+		return ret < 0 ? -1 : ret > 0;
```

Comment:
The return value here is a comparison result (one of -1, 0 or 1). Not explicit returning the three values is pretty confusing.

Reply:
If you like, you can use the `ZEND_NORMALIZE_BOOL()` macro.